pkg/git/git.go

```diff
@@ -182,6 +173,9 @@ func (git Client) Commit(message, user, email string) error {
 
 // Push pushes the changes to the origin remote
 func (git Client) Push() error {
+	if err := git.runGitCmd("config", "push.default", "current"); err != nil {
```

Comment:
Is it possible to avoid having to touch the local configuration for this? I was planning on avoiding that for the user and the email as well so it would be nice to have. (Although no need to block the pr for that)

Reply:
After a couple of experiments this seemed to be the easiest way to cause the push to work without complaints about a missing upstream branch.

The other alternative would be to set the remote tracking branch after checking out, but this did not seem inline with the tactic used for the commit, which applies configuration at the time it is actually required for an action.